tests/backup_restore_test.go

```diff
@@ -0,0 +1,284 @@
+//go:build mysql
+// +build mysql
+
+package tests
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+	"time"
+
+	dbPlugin "github.com/bytebase/bytebase/plugin/db"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/resources/mysql"
+	"github.com/stretchr/testify/require"
+)
+
+// TestBackupRestoreBasic tests basic backup and restore behavior
+// The test plan is:
+// 1. create schema with index and constraint and populate data (TODO(dragonly): add routine/event/trigger)
+// 2. create a full backup
+// 3. clear data
+// 4. restore data
+// 5. validate
+func TestBackupRestoreBasic(t *testing.T) {
+	t.Parallel()
+	a := require.New(t)
+
+	localhost := "127.0.0.1"
+	port := getTestPort(t.Name())
+	username := "root"
+	database := "backup_restore"
+	table := "backup_restore"
+
+	_, stop := mysql.SetupTestInstance(t, port)
+	defer stop()
+
+	db, err := sql.Open("mysql", fmt.Sprintf("root@tcp(localhost:%d)/mysql", port))
+	a.NoError(err)
+	defer db.Close()
+
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", database))
+	a.NoError(err)
+
+	_, err = db.Exec(fmt.Sprintf("USE %s", database))
+	a.NoError(err)
+
+	_, err = db.Exec(fmt.Sprintf(`
+	CREATE TABLE %s (
+		id INT,
+		PRIMARY KEY (id),
+		CHECK (id > 0)
+	);
+	`, table))
+	a.NoError(err)
+
+	const numRecords = 100
+	tx, err := db.Begin()
+	a.NoError(err)
+	defer tx.Rollback()
+	for i := 1; i <= numRecords; i++ {
```

Comment:
Curious why not start with 0, so we don't need to compare numRecords +1 and i

Reply:
Meant to be compatible with gh-ost test. Will change both to start from 0.